src/rules/max-len.ts

```diff
@@ -0,0 +1,207 @@
+import type {
+  Blockquote,
+  Code,
+  Heading,
+  ListItem,
+  Paragraph,
+  TableRow,
+} from "../language/ast-types.ts";
+import { createRule } from "../utils/index.ts";
+import { getTextWidth } from "../utils/text-width.ts";
+
+type OptionObject = {
+  heading?: number;
+  paragraph?: number;
+  listItem?: number;
+  blockquote?: number;
+  tableRow?: number;
+  codeBlock?: number | null;
+  ignoreUrls?: boolean;
+};
+type Option = OptionObject;
+
+const URL_PATTERN =
+  /https?:\/\/(?:w{3}\.)?[\w#%+\-.:=@~]{1,256}\.[()0-9A-Za-z]{1,6}\b[\w#%&()+\-./:=?@~]*/gu;
+
+export default createRule<[Option?]>("max-len", {
+  meta: {
+    type: "layout",
+    docs: {
+      description: "enforce maximum length for various Markdown entities",
+      categories: ["standard"],
+      listCategory: "Layout & Formatting",
+    },
+    fixable: null,
+    hasSuggestions: false,
+    schema: [
+      {
+        type: "object",
+        properties: {
+          heading: {
+            type: "integer",
+            minimum: 1,
+          },
+          paragraph: {
+            type: "integer",
+            minimum: 1,
+          },
+          listItem: {
+            type: "integer",
+            minimum: 1,
+          },
+          blockquote: {
+            type: "integer",
+            minimum: 1,
+          },
+          tableRow: {
+            type: "integer",
+            minimum: 1,
+          },
+          codeBlock: {
+            oneOf: [
+              {
+                type: "integer",
+                minimum: 1,
+              },
+              {
+                type: "null",
+              },
+            ],
+          },
+          ignoreUrls: {
+            type: "boolean",
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+    messages: {
+      maxLenExceeded:
+        "Line length of {{actual}} exceeds the maximum of {{max}}.",
+    },
+  },
+  create(context) {
+    const sourceCode = context.sourceCode;
+
+    const opt = context.options[0] || {};
+    const headingMax = opt.heading ?? 80;
+    const paragraphMax = opt.paragraph ?? 120;
+    const listItemMax = opt.listItem ?? 120;
+    const blockquoteMax = opt.blockquote ?? 120;
+    const tableRowMax = opt.tableRow ?? 120;
+    const codeBlockMax = opt.codeBlock ?? null; // null means ignore
+    const ignoreUrls = opt.ignoreUrls ?? true;
+
+    // Track which ranges we've already checked to avoid double-checking
+    const checkedRanges = new Set<string>();
+
+    /**
+     * Check if a line contains a URL that takes up significant space
+     */
+    function shouldIgnoreLine(text: string, maxLength: number): boolean {
+      if (!ignoreUrls) return false;
+
+      // Find all URLs in the text
+      const urls = [...text.matchAll(URL_PATTERN)];
+      if (urls.length === 0) return false;
+
+      // Calculate the longest URL length
+      const longestUrlLength = Math.max(
```

Comment:
`ignoreUrls` should work the same as `@stylistic/max-len`.

https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/rules/max-len/max-len.ts